engine/src/connector.rs

```diff
@@ -91,12 +89,18 @@ impl<I: IO + Copy> EthConnectorContract<I> {
     }
 
     /// Init eth-connector contract specific data
-    pub fn init_contract(mut io: I, owner_id: AccountId, args: InitCallArgs) {
+    pub fn init_contract(
+        mut io: I,
+        owner_id: AccountId,
+        args: InitCallArgs,
+    ) -> Result<(), error::InitContractError> {
         // Check is it already initialized
-        assert!(
-            !io.storage_has_key(&Self::get_contract_key(&EthConnectorStorageId::Contract)),
-            "ERR_CONTRACT_INITIALIZED"
-        );
+        let contract_key_exists =
+            io.storage_has_key(&Self::get_contract_key(&EthConnectorStorageId::Contract));
+        if contract_key_exists {
+            return Err(error::InitContractError::AlreadyInitialized);
+        }
```

Comment:
Why don't you like factoring out a long-ish function call as a named variable before using it?

Also the present logic is correct; if the exists then we error out because it's already initialized.